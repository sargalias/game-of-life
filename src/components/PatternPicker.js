import React from 'react';

const PatternPicker = ({patterns=[], onChange}) => (
  <div className="pattern-picker-container">
    <select name="" id="" onChange={onChange}>
      {patterns.map((pattern) => <option value={pattern} key={pattern}>{pattern}</option>)}
    </select>
  </div>
);

export default PatternPicker;
