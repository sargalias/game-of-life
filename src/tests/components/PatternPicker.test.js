import React from 'react';
import { shallow } from 'enzyme';
import PatternPicker from '../../components/PatternPicker';


test('PatternPicker should render correctly', () => {
  const wrapper = shallow(<PatternPicker />);
  expect(wrapper).toMatchSnapshot();
});

test('PatternPicker should correctly use its props.patterns', () => {
  const patterns = ['single_cell', 'test', 'test2'];
  const wrapper = shallow(<PatternPicker patterns={patterns} />);
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('select').children().length).toBe(3);
});

test('PatternPicker should correctly make use of its props.onChange', () => {
  const onChange = jest.fn();
  const wrapper = shallow(<PatternPicker onChange={onChange} />);
  wrapper.find('select').simulate('change');
  expect(onChange).toHaveBeenCalled();
});
