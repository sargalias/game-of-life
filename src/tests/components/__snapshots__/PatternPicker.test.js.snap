// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatternPicker should correctly use its props.patterns 1`] = `
<div
  className="pattern-picker-container"
>
  <select
    id=""
    name=""
  >
    <option
      key="single_cell"
      value="single_cell"
    >
      single_cell
    </option>
    <option
      key="test"
      value="test"
    >
      test
    </option>
    <option
      key="test2"
      value="test2"
    >
      test2
    </option>
  </select>
</div>
`;

exports[`PatternPicker should render correctly 1`] = `
<div
  className="pattern-picker-container"
>
  <select
    id=""
    name=""
  />
</div>
`;
